packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -393,13 +399,18 @@ class _GoogleMapOptions {
     this.tiltGesturesEnabled,
     this.trackCameraPosition,
     this.zoomGesturesEnabled,
+    this.liteModeEnabled,
     this.myLocationEnabled,
     this.myLocationButtonEnabled,
     this.padding,
     this.indoorViewEnabled,
     this.trafficEnabled,
     this.buildingsEnabled,
-  });
+  }) {
+    assert(liteModeEnabled == null ||
+        !liteModeEnabled ||
+        (liteModeEnabled && Platform.isAndroid));
```

Comment:
nit: what about leaving this as a noop if `!Platform.isAndroid`?

Reply:
I think we can leave it as is, an assertion helps the user to be more explicit when using liteMode so they are aware it doesn't work on iOS.